lib/web_ui/test/ui/fallback_fonts_golden_test.dart

```diff
@@ -293,6 +309,46 @@ void testMain() {
       await checkDownloadedFamilyForCharCode(0x1f3d5, 'Noto Color Emoji');
     });
 
+    // These code points exist in all of our CJK fonts.
+    final commonCJKCodePoints = <int>[
+      0x56de, // CJK Unified Ideographs
+      0x700b, // CJK Unified Ideographs
+      0x9c57, // CJK Unified Ideographs
+    ];
+
+    for (final codePoint in commonCJKCodePoints) {
+      final codePointHex = '0x${codePoint.toRadixString(16)}';
+      // Simplified Chinese
+      for (final lang in <String>['zh-Hans', 'zh-CN', 'zh-SG', 'zh-MY', 'en-US']) {
+        // Simplified Chinese is also prioritized when preferred language is en-US.
+        test('prioritizes Noto Sans SC for code-point=$codePointHex lang=$lang', () async {
+          await checkDownloadedFamilyForCharCode(codePoint, 'Noto Sans SC', userPreferredLanguage: lang);
+        });
+      }
+
+      // Traditional Chinese
+      for (final lang in <String>['zh-Hant', 'zh-TW', 'zh-MO']) {
+        test('prioritizes Noto Sans TC for code-point=$codePointHex lang=$lang', () async {
+          await checkDownloadedFamilyForCharCode(codePoint, 'Noto Sans TC', userPreferredLanguage: lang);
+        });
+      }
+
+      // Hong Kong
```

Comment:
You could simplify all these tests with:

```dart
const fontFamilyToPreferredLanguage = Map<String, List<String>> {
  'Noto Sans SC': <String>['zh-Hans', 'zh-CN', 'zh-SG'...],
  'Noto Sans TC': <String>['zh-Hant', 'zh-TW', ...],
  ...
}
```

And iterating over its entries to generate all the possible tests without duplicating any code? :)

(It might make the actual `test` harder to find when it fails if we try to search by the error message or line number though)

Reply:
I was actually leaning the opposite direction. I prefer better readability/searchability. I was thinking of removing all `for` loops and doing:
```dart
test('prioritizes Noto Sans TC for code-point=0x700b lang=zh-Hant', () async {
  await checkDownloadedFamilyForCharCode(codePoint, 'Noto Sans TC', userPreferredLanguage: 'zh-Hant');
});
test('prioritizes Noto Sans TC for code-point=0x700b lang=zh-TW', () async {
  await checkDownloadedFamilyForCharCode(codePoint, 'Noto Sans TC', userPreferredLanguage: 'zh-TW');
});
test('prioritizes Noto Sans TC for code-point=0x700b lang=zh-MO', () async {
  await checkDownloadedFamilyForCharCode(codePoint, 'Noto Sans TC', userPreferredLanguage: 'zh-MO');
});
```
The code duplication isn't bad honestly. All the logic lives in `checkDownloadedFamilyForCharCode`.

Now that you suggested testing only one code point, I have an even stronger feeling to remove all the `for` loops 🙂